x/wasm-storage/types/sigma.go

```diff
@@ -0,0 +1,94 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"golang.org/x/exp/constraints"
+)
+
+// Sigma is a 10^6 precision fixed-point unsigned number represented
+// by a uint64, whose last 6 digits represent the fractional part.
+type Sigma uint64
+
+func NewSigma(data []byte) (Sigma, error) {
+	var s Sigma
+	err := binary.Read(bytes.NewReader(data), binary.BigEndian, &s)
+	if err != nil {
+		return s, err
+	}
+	return s, nil
+}
+
+// WholeNumber returns Sigma's whole number part.
```

Comment:
Maybe add a comment, something like `1_000_000` becomes `1` (and same with fractional part). Just for clarity